### PR TITLE
refactor(cleaner): return state from instance generator

### LINKED ISSUE #83: refactor(cleaner): return state from the generator
### Is your feature request related to a problem? Please describe
The instance generator for the cleaner environment currently returns an array containing the initial grid. It should instead returns the full environment state to be consistent with other environments. 

### Describe the solution you'd like
Return an instance of `cleaner.State` in the instance generator.

### PR DESCRIPTION
**Pull request by coyettev**
Closes #83.

### CODE REVIEW

File: jumanji/environments/routing/cleaner/env.py [hunk 1]
```diff
@@ -98,8 +97,10 @@ def __init__(
             render_mode: the mode for visualising the environment, can be "human" or "rgb_array".
             penalty_per_timestep: the penalty returned at each timestep in the reward.
         """
-        self.num_agents = num_agents
-        self.generator = generator or RandomGenerator(num_rows=10, num_cols=10)
+        self.generator = generator or RandomGenerator(
+            num_rows=10, num_cols=10, num_agents=3
```
Comment (dluo96) [inline jumanji/environments/routing/cleaner/env.py:101]: You have to update the docstring to reflect that the default generator takes `num_agents=3`, right?

File: jumanji/environments/routing/cleaner/generator.py [hunk 2]
```diff
@@ -14,44 +14,47 @@
...
-        """Interface for maze generation for the `Cleaner` environment.
+    def __init__(self, num_rows: int, num_cols: int, num_agents: int) -> None:
+        """Interface for instance generation for the `Cleaner` environment.
 
         Args:
-            num_rows: the width of the maze to create.
-            num_cols: the length of the maze to create.
+            num_rows: the width of the grid to create.
+            num_cols: the length of the grid to create.
+            num_agents: number of agents.
```
Comment (dluo96) [inline jumanji/environments/routing/cleaner/generator.py:32]:
```suggestion
            num_agents: the number of agents.
```

File: jumanji/environments/routing/cleaner/types.py [hunk 3]
```diff
@@ -38,7 +38,7 @@ class State:
 
     grid: chex.Array  # (num_rows, num_cols)
     agents_locations: chex.Array  # (num_agent, 2)
-    action_mask: chex.Array  # (num_agent, 4)
+    action_mask: Optional[chex.Array]  # (num_agent, 4)
```
Comment (dluo96) [inline jumanji/environments/routing/cleaner/types.py:41]: NIT: can we rename `num_agent` -> `num_agents` in the comments?

File: jumanji/environments/routing/cleaner/env_test.py [hunk 4]
```diff
@@ -19,15 +19,15 @@
...
+from jumanji.environments.routing.cleaner.generator import Generator
 from jumanji.environments.routing.cleaner.types import Observation, State
 from jumanji.testing.env_not_smoke import check_env_does_not_smoke
 from jumanji.testing.pytrees import assert_is_jax_array_tree
 from jumanji.types import StepType, TimeStep
 
 SAMPLE_GRID = jnp.array(
     [
-        [DIRTY, DIRTY, WALL, DIRTY, DIRTY],
+        [CLEAN, DIRTY, WALL, DIRTY, DIRTY],
```
Comment (dluo96) [inline jumanji/environments/routing/cleaner/env_test.py:30]: This is because the top-left corner is cleaned inside the generator now, right?

File: jumanji/environments/routing/cleaner/generator.py [hunk 5]
```diff
@@ -62,12 +65,30 @@ def __call__(self, key: chex.PRNGKey) -> Maze:
...
+        return State(
+            grid=grid,
+            agents_locations=agents_locations,
+            action_mask=None,
+            step_count=jnp.array(0, jnp.int32),
+            key=state_key,
+        )
 
-    def _adapt_values(self, maze: Maze) -> Maze:
+    def _adapt_values(self, maze: maze_generation.Maze) -> chex.Array:
```
Comment (dluo96) [inline jumanji/environments/routing/cleaner/generator.py:91]: Isn't the return still a `Maze`?

**Review by dluo96 — CHANGES_REQUESTED** [review]
Thank you for your contribution! 🚀 I've left a review 😄

File: jumanji/environments/routing/cleaner/env_test.py [hunk 6]
```diff
@@ -39,17 +39,24 @@
 
 class DummyGenerator(Generator):
     def __init__(self) -> None:
-        super(DummyGenerator, self).__init__(num_rows=5, num_cols=5)
-
-    def __call__(self, key: chex.PRNGKey) -> Maze:
-        return SAMPLE_GRID
+        super(DummyGenerator, self).__init__(num_rows=5, num_cols=5, num_agents=N_AGENT)
```
Comment (clement-bonnet) [inline jumanji/environments/routing/cleaner/env_test.py:42]: Could we state in the class docstring that we are creating a generator of size 5x5 with 3 agents and remove the constant `N_AGENT = 3`?

**Review by clement-bonnet — DISMISSED** [review]
Looks good to me, thanks!

File: jumanji/environments/routing/cleaner/env_test.py (hunk 4 above)
Comment (coyettev) [reply jumanji/environments/routing/cleaner/env_test.py:30]: Yes that's it :+1:

File: jumanji/environments/routing/cleaner/generator.py (hunk 5 above)
Comment (coyettev) [reply jumanji/environments/routing/cleaner/generator.py:91]: It returns what I called a Maze before (which was an alias for chex.Array), however in the state it is typed as a chex.Array, do you think I should create aliases in the state ? It would also mean alias for the action mask and the agent locations to be consistent...

File: jumanji/environments/routing/cleaner/env_test.py (hunk 6 above)
Comment (coyettev) [reply jumanji/environments/routing/cleaner/env_test.py:42]: Sure

File: jumanji/environments/routing/cleaner/generator.py (hunk 5 above)
Comment (dluo96) [reply jumanji/environments/routing/cleaner/generator.py:91]: In that case I would keep the return type as is. Do you think we should keep the type annotation of `maze_generation.Maze` for `maze` or just `chex.Array`?

File: jumanji/environments/routing/cleaner/generator.py [hunk 7]
```diff
@@ -62,12 +65,30 @@ def __call__(self, key: chex.PRNGKey) -> Maze:
...
-        return self._adapt_values(maze)
+        generator_key, state_key = jax.random.split(key)
+        maze = maze_generation.generate_maze(
+            self.num_rows, self.num_cols, generator_key
+        )
+
+        grid = self._adapt_values(maze)
+
+        # Agents start in upper left corner
+        agents_locations = jnp.zeros((self.num_agents, 2), int)
```
Comment (clement-bonnet) [inline jumanji/environments/routing/cleaner/generator.py:78]: NIT
```suggestion
        agents_locations = jnp.zeros((self.num_agents, 2), jnp.int32)
```

**Review by clement-bonnet — APPROVED** [review]
LGTM!

**Review by dluo96 — APPROVED** [review]
LGTM
